pkg/connection/connection.go

```diff
@@ -242,14 +242,14 @@ func (c *csiConnection) ListSnapshots(ctx context.Context, snapshotID string) (*
 
 	rsp, err := client.ListSnapshots(ctx, &req)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
 	if rsp.Entries == nil || len(rsp.Entries) == 0 {
-		return nil, fmt.Errorf("can not find snapshot for snapshotID %s", snapshotID)
+		return nil, 0, fmt.Errorf("can not find snapshot for snapshotID %s", snapshotID)
 	}
 
-	return rsp.Entries[0].Snapshot.Status, nil
+	return rsp.Entries[0].Snapshot.Status, rsp.Entries[0].Snapshot.CreatedAt, nil
```

Comment:
I thought this rsp.Entries[0].Snapshot.CreatedAt is when snapshot is cut, not when it is ready or something else.